crates/wasmtime/src/memory.rs

```diff
@@ -453,7 +453,7 @@ impl Memory {
         let store = store.as_context_mut().0;
         let mem = self.wasmtime_memory(store);
         unsafe {
-            match (*mem).grow(delta, store.limiter()) {
+            match (*mem).grow(delta, store) {
```

Comment:
In addition to this change in this file I think we'll also want to add `Memory::new_async` and `Memory::grow_async` both as `async` methods. That may actually be pretty tricky though because currently the async implementation of `block_on` requires stack-switching and assumes we're on a wasm stack. 

Otherwise adding these methods though would require open-coding a bit the limit logic where we could directly `.await` the future in these functions returned by the limiter. That may be the best route here though? (should ideally only be a small amount of duplication in the runtime implementation to have a method and an `async` method)